File: frontend/src/constants/permissionKeys.js
/**
 * Список всех известных permission_key с описаниями.
 * Соответствует константам в internal/services/permission_keys.go.
 *
 * Naming convention (#187a):
 *   page.<route>                     -- маршруты страниц
 *   tab.<name>                       -- вкладки внутри страниц
 *   component.<name>.<verb>          -- компоненты
 *   action.<verb>.<entity>           -- действия
 *   entity.<name>.<crud>             -- CRUD сущности
 *   table.<slug>.<verb>              -- динамические таблицы (auto-generate)
 *   permission.audit.*               -- журналы и аудит
 *
 * Динамические table.* ключи создаются на бэке при создании таблицы и здесь
 * не перечисляются.
 */
export const ALL_PERMISSION_KEYS = [
  { value: 'page.center', description: 'Страница «Центр заявок»' },
  { value: 'page.cars', description: 'Страница «Автомобили»' },
  { value: 'page.employees', description: 'Страница «Сотрудники»' },
  { value: 'page.statistics', description: 'Страница «Статистика»' },
  { value: 'page.reports', description: 'Страница «Отчёты»' },
  { value: 'page.news', description: 'Страница «Обзор и новости»' },
  { value: 'page.admin', description: 'Раздел «Админка» (общий доступ)' },
  { value: 'page.personal_cabinet', description: 'Личный кабинет' },
  { value: 'page.admin.system_control', description: 'Системное управление (техработы)' },
  { value: 'page.admin.users', description: 'Управление пользователями' },
  { value: 'page.admin.feedback', description: 'Обратная связь' },
  { value: 'page.admin.blacklist', description: 'Чёрный список (машины и люди)' },

  { value: 'entity.cars.read', description: 'Просмотр автомобилей' },
  { value: 'entity.cars.write', description: 'Создание и редактирование автомобилей' },
  { value: 'entity.cars.delete', description: 'Удаление автомобилей' },
  { value: 'entity.employees.read', description: 'Просмотр сотрудников' },
  { value: 'entity.employees.write', description: 'Создание и редактирование сотрудников' },
  { value: 'entity.employees.delete', description: 'Удаление сотрудников' },

  { value: 'action.export.applications', description: 'Экспорт заявок' },
  { value: 'action.approve.application', description: 'Согласование заявок' },
  { value: 'action.forward.application', description: 'Пересылка заявок на согласование' },
  { value: 'action.ban.user', description: 'Блокировка пользователей' },

  { value: 'permission.audit.read', description: 'Просмотр журнала отказов и аудита' },
  { value: 'permission.audit.manage', description: 'Очистка и архивация журнала' },
];

File: internal/services/permission_keys.go
package services

// Permission key naming convention (#187):
//
//   page.<route>                     -- маршруты страниц (page.center, page.cars).
//   tab.<name>                       -- вкладки внутри страниц (tab.applications).
//   component.<name>.<verb>          -- компоненты (component.vehicle_history.export).
//   action.<verb>.<entity>           -- действия (action.delete.employee).
//   entity.<name>.<crud>             -- CRUD сущности (entity.cars.read|write|delete).
//   table.<slug>.<verb>              -- динамические таблицы (генерируются AutoGenerateForTable).
//
// Ключи зашиты как константы здесь и используются в middleware/handler-ах.
// Каталог валидных ключей передаётся фронту через GET /permissions/keys
// для построения дерева в UI.

// Префиксы permission keys (по namespace).
const (
	PrefixPage      = "page."
	PrefixTab       = "tab."
	PrefixComponent = "component."
	PrefixAction    = "action."
	PrefixEntity    = "entity."
	PrefixTable     = "table."
	PrefixAudit     = "permission.audit."
)

// Page-level keys (страницы навигационного меню).
const (
	KeyPageCenter        = "page.center"
	KeyPageCars          = "page.cars"
	KeyPageEmployees     = "page.employees"
	KeyPageStatistics    = "page.statistics"
	KeyPageReports       = "page.reports"
	KeyPageNews          = "page.news"
	KeyPageAdmin         = "page.admin"
	KeyPagePersonal      = "page.personal_cabinet"
	KeyPageSystemControl = "page.admin.system_control"
	KeyPageAdminUsers    = "page.admin.users"
	KeyPageAdminFeedback = "page.admin.feedback"
	KeyPageBlacklist     = "page.admin.blacklist"
)

// Entity-level CRUD keys.
const (
	KeyEntityCarsRead           = "entity.cars.read"
	KeyEntityCarsWrite          = "entity.cars.write"
	KeyEntityCarsDelete         = "entity.cars.delete"
	KeyEntityCarsManualAdd      = "entity.cars.manual_add"
	KeyEntityEmployeesRead      = "entity.employees.read"
	KeyEntityEmployeesWrite     = "entity.employees.write"
	KeyEntityEmployeesDel       = "entity.employees.delete"
	KeyEntityEmployeesManualAdd = "entity.employees.manual_add"
)

// Action-level keys (отдельные действия, не входящие в стандартный CRUD).
const (
	KeyActionExportApplications = "action.export.applications"
	// KeyActionImportList - массовый ввод участников/машин из заполненного Excel-бланка
	// (blank-import). Не super-only: администраторы получают его через adminAll, обычным
	// пользователям выдаётся точечно - импорт обходит форму подачи целиком, поэтому
	// закрыт правом по умолчанию.
	KeyActionImportList         = "action.import.list"
	KeyActionApproveApplication = "action.approve.application"
	KeyActionForwardApplication = "action.forward.application"
	// KeyActionSupplementApplication - дополнить уже поданную заявку (#1685). Право не
	// единственный гейт: сервис всё равно требует, чтобы дополняющий был автором заявки.
	KeyActionSupplementApplication = "action.supplement.application"
	KeyActionBanUser               = "action.ban.user"
)

// Audit-level keys (просмотр и управление журналами).
const (
	KeyAuditRead   = "permission.audit.read"
	KeyAuditManage = "permission.audit.manage"
)

// AllStaticKeys -- список всех известных static keys (без table.* которые auto-generate).
// Используется для валидации входящих запросов и для seed дефолтных групп.
func AllStaticKeys() []string {
	return []string{
		KeyPageCenter,
		KeyPageCars,
		KeyPageEmployees,
		KeyPageStatistics,
		KeyPageReports,
		KeyPageNews,
		KeyPageAdmin,
		KeyPagePersonal,
		KeyPageSystemControl,
		KeyPageAdminUsers,
		KeyPageAdminFeedback,
		KeyPageBlacklist,
		KeyEntityCarsRead,
		KeyEntityCarsWrite,
		KeyEntityCarsDelete,
		KeyEntityCarsManualAdd,
		KeyEntityEmployeesRead,
		KeyEntityEmployeesWrite,
		KeyEntityEmployeesDel,
		KeyEntityEmployeesManualAdd,
		KeyActionExportApplications,
		KeyActionApproveApplication,
		KeyActionForwardApplication,
		KeyActionSupplementApplication,
		KeyActionBanUser,
		KeyAuditRead,
		KeyAuditManage,
	}
}
